File: stack/evaluateReversePolishNotation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        for token in tokens:
            if token == "*" or token == "/" or token == "+" or token == "-":
                operandc, operandb, operanda = 0, stack.pop(), stack.pop()
                if token == "*":
                    operandc = operanda * operandb
                elif token == "/":
                    operandc = operanda // operandb
                    operandc = (operandc + 1) if ((operandc < 0) and (operanda % operandb) != 0) else operandc
                elif token == "+":
                    operandc = operanda + operandb
                else:
                    operandc = operanda - operandb
                stack.append(operandc) 
            else:
                stack.append(int(token))
        return stack.pop()  

File: stack/evaluateReversePolishNotation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        vector<int> stack;
        for(string token: tokens){
            if(token == "+" || token == "-" || token == "*" || token == "/"){
                int operand2 = stack.back();
                stack.pop_back();
                int operand1 = stack.back();
                stack.pop_back();
                int ans;
                if(token == "+")
                    ans = operand1 + operand2;
                else if(token == "-")
                    ans = operand1 - operand2;
                else if(token == "*")
                    ans = operand1 * operand2;
                else
                    ans = operand1 / operand2;
                stack.push_back(ans);
            }
            else
                stack.push_back(stoi(token));
        }
        return stack.back();
    }
};
